### PR TITLE
OSDOCS-4571: Add common terms in authentication and authorization book

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.8+

Issue: https://issues.redhat.com/browse/OSDOCS-4571

Link to docs preview: https://52927--docspreview.netlify.app/openshift-enterprise/latest/authentication/index.html#openshift-monitoring-common-terms_overview-of-authentication-authorization

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by xenolinux** [comment]
@xingxingxia Please take a look at this PR

**Comment by xingxingxia** [comment]
Ack. Will check after some on-hand work.

**Comment by xingxingxia** [comment]
@xenolinux will distribute to my colleague @gkarager to help review.

File: modules/authentication-authorization-common-terms.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+config map::
+A config map provides a way to inject configuration data into pods. You can reference the data stored in a config map in a volume of type `ConfigMap`. Applications running in a pod can use this data.
+
+containers::
+Lightweight and executable images that consist software and all its dependencies. Because containers virtualize the operating system, you can run containers anywhere, from a data center to a public or private cloud to your local host.
+
+Custom Resource (CR)::
+A CR is extension of the Kubernetes API.
+
+Keystone::
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc:32]: acked. added those concepts too

File: modules/authentication-authorization-common-terms.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// * authentication/index.adoc
+
+:_content-type: REFERENCE
+[id="openshift-monitoring-common-terms_{context}"]
+= Common terms
+
+This glossary defines common terms that are used in {product-title} architecture.
+
+authentication::
+To control access to an {product-title} cluster, a cluster administrator can configure user authentication and ensure only approved users access the cluster. To interact with an {product-title} cluster, you must authenticate to the {product-title} API. You can authenticate by providing an OAuth access token or an X.509 client certificate in your requests to the {product-title} API.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc]: modified the description

acked

**Comment by gkarager** [comment]
@xenolinux It would be good to have the authentication basic concepts like Users, Groups, Bearer token, etc.

Also the pod, container, configmap, node all these terms seems unnecessary here as those are common to openshift and they are not specific to auth.

**Comment by xenolinux** [comment]
> @xenolinux It would be good to have the authentication basic concepts like Users, Groups, Bearer token, etc.
> 
> Also the pod, container, configmap, node all these terms seems unnecessary here as those are common to openshift and they are not specific to auth.

As far as the pod, container, config map, and node terms are concerned- I think we should keep them as is. Because when a new customer will visit the authentication book, they also should be able to know common openshift concepts such as pod, node, container, etc

File: modules/authentication-authorization-common-terms.adoc [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+// * authentication/index.adoc
+
+:_content-type: REFERENCE
+[id="openshift-auth-common-terms_{context}"]
+= Common terms
+
+This glossary defines common terms that are used in {product-title} authentication and authorization.
+
+authentication::
+An authentication controls access to an {product-title} cluster and ensures only approved users access the {product-title} cluster.
```
Comment (xingxingxia) [inline modules/authentication-authorization-common-terms.adoc:12]:
```suggestion
An authentication controls access to an {product-title} cluster and ensures only authenticated users access the {product-title} cluster.
```

File: modules/authentication-authorization-common-terms.adoc [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+Mint mode is the default and recommended best practice setting for the Cloud Credential Operator (CCO) to use on the platforms for which it is supported. In this mode, the CCO uses the provided administrator-level cloud credential to create new credentials for components in the cluster with only the specific permissions that are required.
+
+namespaces::
+A namespace isolates specific system resources that are visible to all processes. Inside a namespace, only processes that are members of that namespace can see those resources.
+
+node::
+A worker machine in the {product-title} cluster. A node is either a virtual machine (VM) or a physical machine.
+
+OAuth client::
+OAuth client is used to access a bearer token.
```
Comment (xingxingxia) [inline modules/authentication-authorization-common-terms.adoc:57]:
```suggestion
OAuth client is used to get a bearer token.
```

Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc]: acked

File: modules/authentication-authorization-common-terms.adoc (hunk 3 above)
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc]: acked

**Review by snarayan-redhat — COMMENTED** [review]
This looks great!
Few nits and suggestions. An overall comment would be to try to keep the pattern of definition uniform e.g. 
There is a definition which starts with the term name
`authorization::
Authorization determines whether the identified user has permissions to perform the requested action.` 
and there is another which looks
`node::
A worker machine in the {product-title} cluster. A node is either a virtual machine (VM) or a physical machine.`

File: modules/authentication-authorization-common-terms.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc]: acked

File: modules/authentication-authorization-common-terms.adoc [hunk 5]
```diff
@@ -0,0 +1,87 @@
+// Module included in the following assemblies:
+//
+// * authentication/index.adoc
+
+:_content-type: REFERENCE
+[id="openshift-auth-common-terms_{context}"]
+= Common terms
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc:7]: acked

File: modules/authentication-authorization-common-terms.adoc [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+Bearer token is used to authenticate to API with the header `Authorization: Bearer <token>`.
+
+Cloud Credential Operator::
+The Cloud Credential Operator (CCO) manages cloud provider credentials as custom resource definitions (CRDs).
+
+config map::
+A config map provides a way to inject configuration data into pods. You can reference the data stored in a config map in a volume of type `ConfigMap`. Applications running in a pod can use this data.
+
+containers::
+Lightweight and executable images that consist software and all its dependencies. Because containers virtualize the operating system, you can run containers anywhere, from a data center to a public or private cloud to your local host.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc]: acked

File: modules/authentication-authorization-common-terms.adoc [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+Custom Resource (CR)::
+A CR is extension of the Kubernetes API.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc]: acked

File: modules/authentication-authorization-common-terms.adoc [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+groups::
+A group is a set of users. A group is useful for granting permissions to multiple users one time.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc:33]: acked

File: modules/authentication-authorization-common-terms.adoc [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+LDAP is a protocol that queries user information.
+
+manual mode::
+In manual mode, a user manages cloud credentials instead of the Cloud Credential Operator (CCO).
+
+mint mode::
+Mint mode is the default and recommended best practice setting for the Cloud Credential Operator (CCO) to use on the platforms for which it is supported. In this mode, the CCO uses the provided administrator-level cloud credential to create new credentials for components in the cluster with only the specific permissions that are required.
+
+namespaces::
+A namespace isolates specific system resources that are visible to all processes. Inside a namespace, only processes that are members of that namespace can see those resources.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc:51]: acked

File: modules/authentication-authorization-common-terms.adoc [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+A pod is the smallest logical unit in Kubernetes. A pod is comprised of one or more containers to run in a worker node.
+
+regular users::
+Users that are created automatically in the cluster upon first login or via the API.
+
+request header::
+A request header is an HTTP header that is used to provide information about HTTP request context, so that the server can track the response of the request.
+
+role-based access control (RBAC)::
+A key security control to ensure that cluster users and workloads have only access to resources required to execute their roles.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc]: acked

File: modules/authentication-authorization-common-terms.adoc [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+service accounts::
+Serivce accounts are used by cluster components or applications.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/authentication-authorization-common-terms.adoc]: acked

**Pull request merged**